skopt/optimizer/optimizer.py

```diff
@@ -176,7 +186,101 @@ def _check_arguments(self, base_estimator, n_random_starts, acq_optimizer):
                              "'sampling', got {0}".format(acq_optimizer))
         self.acq_optimizer = acq_optimizer
 
-    def ask(self):
+    def copy(self, copy_random_state=True):
+        """Creates a shallow copy of optimizer."""
+
+        params = {
+            'dimensions':self.space.dimensions,
+            'base_estimator':self.base_estimator,
+            'n_random_starts':self._n_random_starts,
+            'acq_func':self.acq_func,
+            'acq_optimizer':self.acq_optimizer,
+            'acq_func_kwargs':self.acq_func_kwargs,
+            'acq_optimizer_kwargs':self.acq_optimizer_kwargs,
+        }
+
+        if copy_random_state:
+            params['random_state'] = copy.copy(self.rng)
+
+        optimizer = Optimizer(**params)
+
+        if hasattr(self, "gains_"):
+            optimizer.gains_ = np.copy(self.gains_)
+
+        if len(self.Xi) > 0:
```

Comment:
Indeed it is nicer. I changed it here and in other parts of the code where I was able to find similar thing.